text/0095-cognito-construct-library/working-backwards-readme.md

```diff
@@ -0,0 +1,567 @@
+    replyTo: 'support@my-awesome-app.com'
+  }
+});
+```
+
+> Internal Note: configured via UserPool-EmailConfiguration
+
+### Triggers
+
+User pools can be configured with a number of lambda function backed triggers. They are available in the CDK via the
+`triggers` property. [Go
+here](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-identity-pools-working-with-aws-lambda-triggers.html)
+to read more about user pool workflows using lambda triggers, and details around each trigger.
+
+Check out the [documentation](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-cognito/lib/user-pool.ts#L148)
+to find the list of triggers supported by the CDK.
+
+Code sample:
+
+```ts
+const presignupLambda = new lambda.Function(this, 'presignup' {
+  // ...
+});
+
+const postsignupLambda = new lambda.Function(this, 'postsignup', {
+  // ...
+});
+
+new UserPool(this, 'myuserpool', {
+  // ...
+  // ...
+  triggers: {
+    preSignUp: presignupLambda,
+    postSignUp: postsignupLambda,
+    preAuthentication: preauthLambda,
+    // ...
+  }
+});
+```
+
+Triggers can also be added to the user pool outside of its constructor. This can be done via individual methods such as
+`addPreSgnUpTrigger()`, `addPostSignUpTrigger()`, `addPreAuthentication()`, etc.
+
+### Importing User Pools
+
+Any user pool that has been created outside of this stack, can be imported into the CDK app. Importing a user pool
+allows for it to be used in other parts of the CDK app that reference an `IUserPool`. However, imported user pools have
+limited configurability. As a rule of thumb, none of the properties that is are part of the
+[`AWS::Cognito::UserPool`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-cognito-userpool.html)
+CloudFormation resource can be configured.
+
+User pools can be imported either by specifying the user pool id, via the `UserPool.fromUserPoolId()` API, or by
+specifying the user pool ARN, via the `UserPool.fromUserPoolArn()` API.
+
+```ts
+const stack = new Stack(app, 'my-stack');
+
+const awesomePool = UserPool.fromUserPoolId(stack, 'awesome-user-pool', 'us-east-1_oiuR12Abd');
+
+const otherAwesomePool = UserPool.fromUserPoolArn(stack, 'other-awesome-user-pool',
+  'arn:aws:cognito-idp:eu-west-1:123456789012:userpool/us-east-1_mtRyYQ14D');
+```
+
+### Users
+
+Users are added to pools either as part of sign up or via the Cognito APIs directly. In addition to these, it is
+possible to add users to the user pool via the CDK.
+
+Use the `createUser()` API on the user pool to add users to the pool, as so -
+
+```ts
+const userpool = new UserPool(this, 'myuserpool', {
+  // ...
+  // ...
+});
+
+userpool.createUser(this, 'myuserpool-supportuser', {
+  userName: 'support-user',
+  deliveryMediums: [ ContactMethods.EMAIL, ContactMethods.PHONE ],
+  attributes: [
+    { name: StandardAttrs.EMAIL, value: 'support@awesomeapp.com' },
+    { name: 'callingcode' value: '12' }
```

Comment:
Missing comma after 'callingcode'.

Reply:
At this point I would have a used a map:

```ts

userpool.createUser(this, 'myuserpool-supportuser', {
  attributes: {
    [StandardAttrs.EMAIL]: 'support@awesomeapp.com',
    callingcode: 12
  }
});

Reply:
Good call. Changed.